### ISSUE TITLE
Navigation showing duplicating results when multilingual page and requesting non-default lang

### ISSUE THREAD

**Issue by erickgonzalez** [dotCMS/core#23890]
### Problem Statement

If a customer has multilingual pages or file assets with `show on menu = true` and requests the navigation for the non-default language, we display the one in the requested language but also the one of the default language.

### Steps to Reproduce

1- Create a folder named e.g `ca` with show on menu enabled.
2- Create a page in the default language with show on menu enabled.
3- Create a new version of the page in another language, e.g Spanish (id=2).
4- Add a Code Snippet Widget with the following code:
```
#set($navRootLevel = $navtool.getNav("/ca/", 2))
#foreach($primaryLevel in $navRootLevel)
-----</br>
$primaryLevel.title</br>
#foreach($secondaryLevel in $primaryLevel.children)
$secondaryLevel.title</br>
#end
#end
```
5- Both pages are shown, but we're only requesting the nav for the Spanish language.

**Workaround:**
Filter by language Id the results of the navtool
```
#set($navRootLevel = $navtool.getNav("/ca/", 2))
#foreach($primaryLevel in $navRootLevel)
#if($primaryLevel.languageId == 2)
-----</br>
$primaryLevel.title</br>
#end
#foreach($secondaryLevel in $primaryLevel.children)
#if($secondaryLevel.languageId == 2)
$secondaryLevel.title</br>
#end
#end
#end
```

### Acceptance Criteria

If the page or file asset exists in the requested language should be shown in this, and only show the default language in case it doesn't exist in the requested language.

### dotCMS Version

Introduced in 22.02.

Tested in 22.03 LTS and 23.01 LTS.

### Proposed Objective

Code Maintenance

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/109808

**Comment by erickgonzalez** [comment]
Not reproducible in 21.06 LTS

**Comment by bryanboza** [comment]
Fixed, tested on release-23.02 // Docker // FF 

New related case moved to a card #24059
